### ISSUE TITLE
Error when passing baseUrl with value null or undefined to cypress.run

### ISSUE THREAD

**Issue by guilhermebruzzi** [cypress-io/cypress#2160]
### Current behavior:

When you pass a config to cypress.run and it contains the `baseUrl` param with the value `null` (which is its default value: https://docs.cypress.io/guides/references/configuration.html#Global) or `undefined` it throws the error: 

Expected 'baseUrl' to be a fully qualified URL (starting with http:// or https://). Instead the value was: "null"

### Desired behavior:

Should execute the tests just like if you didn't pass the `baseUrl` param.

### Steps to reproduce:

This is a code similar to the one I tried to run:

```js
const cypress = require('cypress')

cypress
      .run({
        headed: false,
        watchForFileChanges: false,
        config: {
          record: false,
          video: false,
          baseUrl: null, // <- the problem
        },
      })
```

### Versions

cypress: 3.0.2 
Operating system: macOS High Sierra (10.13.5)

**Comment by jennifer-shehane** [comment]
Can you not omit the `baseUrl` key from the `config` above when you run `cypress.run`? I'm trying to understand why passing `null` is unavoidable here. Is it always passed as `null` if you pass any other config value?

I feel like erroring would be the desired behavior for most of our users, as they may be generating the baseUrl and would not get notification of an incorrect baseUrl until a later, less obvious point in running their tests.

**Comment by guilhermebruzzi** [comment]
Hi @jennifer-shehane, that's what I did to fix it, I didn't include the `baseUrl` key :D

But I still think it should be fixed, because in my case and I believe in a lot of cases (that you build the cypress config dynamically), it looks better to do:

```js
const cypressConfig = {
  baseUrl: config.host ? `http://${config.host}:${config.port}` : null,
  ...config,
}
```

Than doing this:

```js
const cypressConfig = config.host
    ? {
        baseUrl: `http://${config.host}:${config.port}`,
        ...config,
      }
    : config
```

And by looking at the cypress code it looks clear that it wasn't an intended behavior: https://github.com/cypress-io/cypress/blob/39164a5c31c87f8b2e75e4bc3910ab00ca558a19/packages/server/lib/config.coffee#L193

Because looking at the previous code, cypress only validate a param if it isn't the default value. So I believe the problem is that it is converting any baseUrl value to a string (couldn't find where).

If you look at the message: `Expected 'baseUrl' to be a fully qualified URL (starting with http:// or https://). Instead the value was: "null"` It makes me believe that when I pass `{ baseUrl: null, ...}` it converts to `{ baseUrl: "null" }`

Do you know where this conversion could be happening?

**Comment by jennifer-shehane** [comment]
Yeah, I can see your points for wanting the conditional statement here.

Well, this error could be thrown in a few different places - as we do some verification later when doing `visit` or `request` with an invalid `baseUrl`. 

Where is this error actually being thrown? Is it displaying in the terminal? In the desktop before you can see your list of specs? Or as your tests begin running.

**Comment by guilhermebruzzi** [comment]
@jennifer-shehane It display in the terminal before any spec is considered to be running, the full Cypress output is:

```
We found an invalid configuration value:

Expected 'baseUrl' to be a fully qualified URL (starting with http:// or https://). Instead the value was: "null"
error Command failed with exit code 1.
```

**Comment by jiangpingshendd01** [comment]
hi @jennifer-shehane  @guilhermebruzzi  I meet the same issue today ,and no matter how to edit the config, cypress always throw the same error? would u mind tell me how to deal with it ?
<img width="647" alt="屏幕快照 2020-01-22 下午6 24 35" src="https://user-images.githubusercontent.com/53250925/72886284-7ba98880-3d44-11ea-8b51-30f55c839d3d.png">

**Comment by guilhermebruzzi** [comment]
@jiangpingshendd01 I think the problem is where you put the baseUrl in your case.

See my steps to reproduce above:
```js
This is a code similar to the one I tried to run:

const cypress = require('cypress')

cypress
      .run({
        headed: false,
        watchForFileChanges: false,
        config: {
          record: false,
          video: false,
          baseUrl: null, // <- the problem
        },
      })
```

the baseUrl param is inside a "config" key, so in your case you should probably do: `myjson.config.baseUrl = "https://dev-my.hktester.com"`
